controllers/pattern_controller.go

```diff
@@ -504,3 +534,106 @@ func (r *PatternReconciler) actionPerformed(p *api.Pattern, reason string, err e
 	}
 	return r.onReconcileErrorWithRequeue(p, reason, err, nil)
 }
+
+// MBP-184 Implementation
+// applyPatternAppDetails retrieves the status for the Pattern applications created in ArgoCD.
+// Once the Application Status is retrieve it appends the information
+// to the array and the CR is updated.
+func (r *PatternReconciler) applyPatternAppDetails(client argoclient.Interface, input *api.Pattern) (*api.Pattern, error) {
+	output := input.DeepCopy()
+	var applicationInfo api.PatternApplicationInfo
+	var labelFilter = "validatedpatterns.io/pattern=" + output.ObjectMeta.Name
+
+	// Retrieving all applications that contain the label
+	// oc get Applications -A -l validatedpatterns.io/pattern=<pattern-name>
+	//
+	// The VP framework adds the label to each application it creates.
+	applications, err := client.ArgoprojV1alpha1().Applications("").List(context.Background(),
+		metav1.ListOptions{
+			LabelSelector: labelFilter,
+		})
+	if err != nil {
+		//fmt.Println("Did not find Applications", err)
+		return nil, err
+	}
+
+	// Reset the array
+	if len(applications.Items) > 0 {
+		output.Status.Applications = nil
+	}
+
+	// Loop through the Pattern Applications and append the details to the Applications array
+	for _, app := range applications.Items {
+		// fmt.Println(app.Name, " ==> [", app.Status.Health.Status, "] == [", app.Status.Sync.Status, "]")
+		// Add Application information to ApplicationInfo struct
+		applicationInfo.Name = app.Name
+		applicationInfo.Namespace = app.Namespace
+		applicationInfo.AppHealthStatus = string(app.Status.Health.Status)
+		applicationInfo.AppHealthMessage = app.Status.Health.Message
+		applicationInfo.AppSyncStatus = string(app.Status.Sync.Status)
+
+		// Now let's append the Application Information
+		output.Status.Applications = append(output.Status.Applications, *applicationInfo.DeepCopy())
+	}
+	return output, err
+}
+
+// MBP-184 Implementation
+// updatePatternCRDetails compares the current CR Status.Applications array
+// against the instance.Status.Applications array.
+// Returns true if the CR was updated else it returns false
+func (r *PatternReconciler) updatePatternCRDetails(input *api.Pattern) (bool, error) {
+	fUpdateCR := false
+
+	// Update CR for the Pattern Instance
+	if input.Status.Applications != nil {
+		// Create an empty Pattern CR
+		patternCR := &api.Pattern{}
+
+		// Get the current Pattern CR
+		_ = r.Client.Get(context.TODO(), Client.ObjectKey{
+			Namespace: "openshift-operators",
+			Name:      input.ObjectMeta.Name}, patternCR)
+
+		// Check to see if the Pattern CR has a list of Applications
+		// If it doesn't and we have a list of Applications
+		// Let's update the Pattern CR and set the update flag to true
+		if len(patternCR.Status.Applications) <= 0 && len(input.Status.Applications) > 0 {
```

Comment:
This bit is a little unclear to me because I don't think you have any guarantees that the ordering of the input applications are the same as the existing ones?